### PR TITLE
Add configuration to ignore /ok-to-test comment

### LINKED ISSUE #10118: ok-to-test should be reverted for new commits
Currently, once someone from the trusted org comments `ok-to-test` to a PR, any upcoming commit will trigger the jobs. This could lead to security issues if the commits after `ok-to-test` comment introduce malicious code.

### PR DESCRIPTION
**Pull request by matthyx**
Fixes #10118

### CODE REVIEW

File: prow/plugins/trigger/pull-request.go [hunk 1]
```diff
@@ -137,7 +137,21 @@ func buildAllIfTrusted(c client, trigger *plugins.Trigger, pr github.PullRequest
 }
 
 func welcomeMsg(ghc githubClient, trigger *plugins.Trigger, pr github.PullRequest) error {
-	commentTemplate := `Hi @%s. Thanks for your PR.
+	var commentTemplate string
+	if trigger.IgnoreOkToTest {
+		commentTemplate = `Hi @%s. Thanks for your PR.
+
+Testing PRs from untrusted members has been disabled in this repo, however collaborators can trigger them using ` + "`/test *`" + `.
```
Comment (cjwagner) [inline prow/plugins/trigger/pull-request.go]:
```suggestion
PRs from untrusted users cannot be marked as trusted with `/ok-to-test` in this repo meaning untrusted PR authors can never trigger tests themselves. Collaborators can still trigger tests on the PR using ` + "`/test all`" + `.
```

File: prow/plugins/trigger/pull-request.go [hunk 2]
```diff
@@ -137,7 +137,21 @@ func buildAllIfTrusted(c client, trigger *plugins.Trigger, pr github.PullRequest
 }
 
 func welcomeMsg(ghc githubClient, trigger *plugins.Trigger, pr github.PullRequest) error {
-	commentTemplate := `Hi @%s. Thanks for your PR.
+	var commentTemplate string
+	if trigger.IgnoreOkToTest {
+		commentTemplate = `Hi @%s. Thanks for your PR.
```
Comment (cjwagner) [inline prow/plugins/trigger/pull-request.go]: This template requires different parameters from the original template, but is still created with the original parameters on line 181 so this won't format properly.

**Comment by matthyx** [comment]
> Should we avoid adding the needs-ok-to-test label in repos where /ok-to-test is disabled or do we still need to apply the label in order for triggers new logic to work?

Actually `needs-ok-to-test` is purely cosmetic now (well, and maybe used in queries and reports to identify PRs that require members to allow testing) so it doesn't make sense to add it in this case, good point.

File: prow/plugins/trigger/generic-comment.go [hunk 3]
```diff
@@ -50,29 +50,32 @@ func handleGenericComment(c client, trigger *plugins.Trigger, gc github.GenericC
...
-		// Check for the presence of the needs-ok-to-test label and remove it
-		// if a trusted member has commented "/ok-to-test".
+		// If a trusted member has commented "/ok-to-test",
+		// eventually add ok-to-test and remove needs-ok-to-test.
 		l, err := c.GitHubClient.GetIssueLabels(org, repo, number)
 		if err != nil {
 			return err
 		}
-		if isOkToTest && github.HasLabel(labels.NeedsOkToTest, l) {
+		if isOkToTest && !github.HasLabel(labels.OkToTest, l) {
```
Comment (matthyx) [inline prow/plugins/trigger/generic-comment.go]: Nit: actually the source of trust is `ok-to-test`.

**Pull request merged**